hack/generator/pkg/astmodel/testdata/SetMapOfOptionalFromMapOfRequired.golden

```diff
@@ -33,13 +33,11 @@ func (person Person) ConvertToVNext(destination *vNext.Person) error {
 	for ratingKey, ratingValue := range person.ratings {
 		// Shadow the loop variable to avoid aliasing
 		ratingValue := ratingValue
-		var rating int
 		if ratingValue != nil {
-			rating = *ratingValue
+			ratingMap[ratingKey] = *ratingValue
 		} else {
-			rating = 0
+			ratingMap[ratingKey] = 0
```

Comment:
This is weird, but I *think* it's right - the source map has a key that maps to nil, that should convert  to a key with a zero value. Someone could argue that it should convert to a missing key?

Reply:
I'm pretty sure this is right - we're maintaining the fact that the key value did exist in the original map.

Reply:
Do we need to stash the "nil-ness" of a particular value in the bag and retrieve it on the way back? That seems tricky but would:
1. Maintain roundtripping exactly.
2. Push the default behavior to be that nil values are "dropped" (shunted to the bag).

Both nil values being dropped and nil values being the zero value of the type have a chance of being wrong (so we'd need to write an override in those cases I think?)

I think this same round-tripping problem appears for all optional types (so optional arrays and plain optional values also have the same problem?)

Reply:
We don't (yet!) have a property bag, but it would certainly be possible to use it to tunnel extra information through. I've added this to Issue #353

Reply:
Yeah, anytime there's a transition between optional and non-optional we have this problem. It occurs both ways in ARM APIs - optional things that become manditory, and mandatory things that become optional. Stashing additional information in the property bag sounds like a good solution.

Reply:
Might be worth leaving a note in the code as well (TODO) just as a reminder to do this eventually, although I suppose eventually we're going to have roundtripping tests as well that would notice it in any case.

Reply:
I've added this to Issue #353 so I think that's enough.